### PR TITLE
New storage performance scripts

### PR DESCRIPTION
**Pull request by dulili**

### CODE REVIEW

File: testscripts/rdma_check_Ibstat.ps1 [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+.Description
+       <test>
+            <testName>rdma_check_Ibstat</testName>
+            <testID>ESX-RDMA-004</testID>
+            <setupScript>
+                <file>setupscripts\add_pvrdma.ps1</file>
+            </setupScript>
+            <testScript>testscripts\rdma_check_Ibstat.ps1</testScript>
+            <testParams>
+                <param>TC_COVERED=RHEL-111208</param>
```
Comment (yangboroy) [inline testscripts/rdma_check_Ibstat.ps1]: add RHEL 6 TC_covered ID.

File: remote-scripts/stor_performance_fio.sh [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+##
+## Main
+##
+###############################################################################
+#Check the guest version which use python2.x, so install pip
+if [[ $DISTRO != "redhat_8" ]]; then
+    yum install -y python-yaml
+    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+    python get-pip.py
+    pip install click pandas numpy scipy
```
Comment (yangboroy) [inline remote-scripts/stor_performance_fio.sh]: add comment for who use it.

File: remote-scripts/stor_performance_fio.sh [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    UpdateSummary "For rhel6 and rhel7, python2.x use pip"
+else
+    yum install python3-yaml -y
+    pip3 install click pandas numpy scipy   
+    #For RHEL8, manually create soft link python point to python3.
+    ln -s /usr/bin/python3 /usr/bin/python
+    UpdateSummary "For rhel8, use pip3 install modules"
+fi
+
+#pip install required python modules.
```
Comment (yangboroy) [inline remote-scripts/stor_performance_fio.sh]: update comments.

File: remote-scripts/stor_performance_fio.sh [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    UpdateSummary "Test failed.Test disk /dev/$disk not exist."
+    SetTestStateAborted
+    exit 1
+else
+    LogMsg " Test disk /dev/$disk exist."
+    UpdateSummary "Test disk /dev/$disk exist."
+fi
+
+# Do Partition for Test disk if needed.
+if [[ $FS != raw ]];then
```
Comment (yangboroy) [inline remote-scripts/stor_performance_fio.sh]: change commend to one line.

File: remote-scripts/stor_performance_fio.sh [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    # Get new partition
+    kpartx /dev/$disk
+
+    # Wait a while
+    sleep 6
+
+    # Format with file system
+    if [[ $DiskType = NVMe ]]; then
+        mkfs.$FS /dev/"$disk"p1
```
Comment (yangboroy) [inline remote-scripts/stor_performance_fio.sh]: change $disk p1 to one value

File: remote-scripts/stor_performance_fio.sh [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+EOF
+
+    # Get new partition
+    kpartx /dev/$disk
+
+    # Wait a while
+    sleep 6
+
+    # Format with file system
+    if [[ $DiskType = NVMe ]]; then
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: part="/dev/{$disk}p1"

Comment (SCHEN2015) [reply remote-scripts/stor_performance_fio.sh]: line 106-120 duplicated with 123-137.

File: remote-scripts/stor_performance_fio.sh [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+mkdir -p $path
+
+#Download fio python scripts from github.
+cd /root
+git clone https://github.com/SCHEN2015/virt-perf-scripts.git
+cd /root/virt-perf-scripts/block
+
+# Execute fio test
+if [[ $FS = raw ]]; then
+    ./RunFioTest.py --rounds 1 --backend $DiskType --driver $DiskType --fs $FS --filename /dev/$disk --log_path $path
```
Comment (yangboroy) [inline remote-scripts/stor_performance_fio.sh]: new add one value for backend.

Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: Add a new variable here.

File: remote-scripts/stor_performance_fio.sh [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+    if [ $? -ne 0 ]; then
+        LogMsg "Test Failed. RunFioTest.py --rounds 1 --backend $DiskType --driver $DiskType --fs $FS --filename /dev/$disk --log_path $path fio run for RAW disk failed.$path"
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: extract the code out.

File: remote-scripts/stor_performance_fio.sh [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+        exit 1
+    else
+       LogMsg " fio run for FS disk successfully."
+       UpdateSummary "fio run for FS disk successfully."
+    fi
+fi
+    
+# Generate Fio test report
+./GenerateTestReport.py --result_path $path
+file=`ls $path`
```
Comment (yangboroy) [inline remote-scripts/stor_performance_fio.sh]: debug info, delete it.

File: remote-scripts/stor_performance_fio.sh [hunk 10]
```diff
@@ -12,81 +12,89 @@
...
 ###############################################################################
 #Check the guest version which use python2.x, so install pip
+#Below moudle yaml,click pandas numpy scipy is used by python script RunFioTest.py and GenerateTestReport.py.
 if [[ $DISTRO != "redhat_8" ]]; then
-    yum install -y python-yaml
-    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-    python get-pip.py
-    pip install click pandas numpy scipy
-    UpdateSummary "For rhel6 and rhel7, python2.x use pip"
+	yum install -y python-yaml
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: I found this one can be install with `pip` the name should be `PyYAML`, you can check with `pip freeze`.

File: remote-scripts/stor_performance_fio.sh [hunk 11]
```diff
@@ -12,81 +12,89 @@
...
+	python get-pip.py
+	pip install click pandas numpy scipy
+	UpdateSummary "For rhel6 and rhel7, python2.x use pip"
 else
-    yum install python3-yaml -y
-    pip3 install click pandas numpy scipy   
-    #For RHEL8, manually create soft link python point to python3.
-    ln -s /usr/bin/python3 /usr/bin/python
-    UpdateSummary "For rhel8, use pip3 install modules"
+	yum install python3-yaml -y
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: The same

File: remote-scripts/stor_performance_fio.sh [hunk 12]
```diff
@@ -151,43 +145,44 @@ git clone https://github.com/SCHEN2015/virt-perf-scripts.git
...
-        SetTestStateFailed
-        exit 1
-    else
-       LogMsg " fio run for FS disk successfully."
-       UpdateSummary "fio run for FS disk successfully."
-    fi
+	filename="/test/test"
 fi
-    
+./RunFioTest.py --rounds 1 --backend $backend --driver $DiskType --fs $FS --filename $filename --log_path $path
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: Remove `--rounds 1` for real testing.

File: remote-scripts/stor_performance_fio.sh [hunk 13]
```diff
@@ -151,43 +145,44 @@ git clone https://github.com/SCHEN2015/virt-perf-scripts.git
...
+if [ $? -ne 0 ]; then
+	LogMsg "Test Failed. fio run failed.$path"
+	UpdateSummary "Test failed.fio run failed.$path,./RunFioTest.py --rounds 1 --backend $backend --driver $DiskType --fs $FS --filename $filename --log_path $path"
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: The same.

File: remote-scripts/stor_performance_fio.sh [hunk 14]
```diff
@@ -151,43 +145,44 @@ git clone https://github.com/SCHEN2015/virt-perf-scripts.git
...
+	LogMsg " fio run successfully."
+	UpdateSummary "fio run successfully."
+fi
+
 # Generate Fio test report
 ./GenerateTestReport.py --result_path $path
-file=`ls $path`
+if [ $? -ne 0 ]; then
+	LogMsg "Test report generate failed"
+	UpdateSummary "Test report generate failed,$path and $file"
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: General comments: add ` ` (a blank) before `$path`... :100:

File: remote-scripts/stor_performance_fio.sh [hunk 15]
```diff
@@ -152,8 +152,8 @@ else
 fi
 ./RunFioTest.py --backend $backend --driver $DiskType --fs $FS --filename $filename --log_path $path
 if [ $? -ne 0 ]; then
-	LogMsg "Test Failed. fio run failed.$path"
-	UpdateSummary "Test failed.fio run failed.$path,./RunFioTest.py --rounds 1 --backend $backend --driver $DiskType --fs $FS --filename $filename --log_path $path"
+	LogMsg "Test Failed. fio run failed."
+	UpdateSummary "Test failed.fio run failed. RunFioTest.py --rounds 1 --backend $backend --driver $DiskType --fs $FS --filename $filename --log_path $path"
```
Comment (SCHEN2015) [inline remote-scripts/stor_performance_fio.sh]: remove `--rounds 1` for real testing

**Pull request merged**